src/hash.cr

```diff
@@ -966,7 +1559,7 @@ class Hash(K, V)
   # hash_a # => {"foobar" => {"foo" => "bar"}}
   # ```
   def clone
-    hash = Hash(K, V).new(initial_capacity: @buckets_size)
+    hash = Hash(K, V).new(initial_capacity: @size)
```

Comment:
clone could be extremely optimized (in case when `@deleted_count < @size`): we need just copy entries and indices, no need in iteration.

Reply:
You are right! I think a similar optimization could have been done in the previous code. I'd like to leave this optimization for a separate PR so this PR doesn't become a bit harder to review (and get right 😅).